pkg/test/fuzz/generate.go

```diff
@@ -0,0 +1,316 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fuzz
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+)
+
+// IDEA: Load all the samples, and check that we have all the KRM paths covered
+
+func FillWithRandom(t *testing.T, randStream *rand.Rand, msg proto.Message) {
+	fillWithRandom0(t, randStream, msg.ProtoReflect())
+}
+
+func fillWithRandom0(t *testing.T, randStream *rand.Rand, msg protoreflect.Message) {
+	descriptor := msg.Descriptor()
+	if string(descriptor.FullName()) == "google.protobuf.Duration" {
+		count := randStream.Intn(10)
+		// Bias to zero
+		if count > 4 {
+			return
+		}
+		seconds := randStream.Intn(365 * 24 * 60 * 60)
+		nanos := randStream.Intn(1000000000)
+		msg.Set(descriptor.Fields().ByName("seconds"), protoreflect.ValueOfInt32(int32(seconds)))
+		msg.Set(descriptor.Fields().ByName("nanos"), protoreflect.ValueOfInt32(int32(nanos)))
+		return
+	}
+
+	fields := descriptor.Fields()
+	n := fields.Len()
+	for i := 0; i < n; i++ {
+		field := fields.Get(i)
+
+		if field.IsList() {
+			count := randStream.Intn(10)
+			// Bias heavily to zero
+			if count > 4 {
+				count = 0
+			}
+			listVal := msg.Mutable(field).List()
+			switch field.Kind() {
+			case protoreflect.MessageKind:
+				for j := 0; j < count; j++ {
+					el := listVal.AppendMutable()
+					fillWithRandom0(t, randStream, el.Message())
+				}
+			case protoreflect.StringKind:
+				for j := 0; j < count; j++ {
+					s := randomString(randStream)
+					listVal.Append(protoreflect.ValueOf(s))
+				}
+
+			default:
+				t.Fatalf("unhandled field kind %v: %v", field.Kind(), field)
+			}
+			continue
+		}
+
+		if field.IsMap() {
+			count := randStream.Intn(10)
+			// Bias heavily to zero
+			if count > 4 {
+				count = 0
+			}
+			mapType := fmt.Sprintf("%s->%s", field.MapKey().Kind(), field.MapValue().Kind())
+			switch mapType {
+			case "string->string":
+				mapVal := msg.Mutable(field).Map()
+				for j := 0; j < count; j++ {
+					k := randomString(randStream)
+					v := randomString(randStream)
+					mapVal.Set(protoreflect.ValueOf(k).MapKey(), protoreflect.ValueOf(v))
+				}
+
+			default:
+				t.Fatalf("unhandled map kind %q: %v", mapType, field)
+			}
+			continue
+		}
+
+		if field.Cardinality() == protoreflect.Optional {
+			if randStream.Intn(3) < 2 {
+				continue
+			}
+		}
+
+		switch field.Kind() {
+		case protoreflect.MessageKind:
+			fieldVal := msg.Mutable(field)
+			fillWithRandom0(t, randStream, fieldVal.Message())
+
+		case protoreflect.BoolKind:
+			msg.Set(field, protoreflect.ValueOfBool(randStream.Intn(2) == 1))
+
+		case protoreflect.DoubleKind:
+			msg.Set(field, protoreflect.ValueOfFloat64(randStream.NormFloat64()))
+		case protoreflect.Int32Kind:
```

Comment:
I feel like some resource may also use `UInt32Kind` and `UInt64Kind`

Reply:
Yes, but we will t.Fatalf when we hit one :-)